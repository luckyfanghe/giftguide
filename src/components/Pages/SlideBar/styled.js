import styled from "styled-components";

export const Body = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  border: ${props => (props.border ? 1 : 0)}px solid black;
`;

export const Close = styled.div`
  display: flex;
  justify-content: flex-end;
  font-size: 40px;
  border: ${props => (props.border ? 1 : 0)}px solid black;
`;

export const Logo = styled.img.attrs(props => ({
  alt: props.alt,
  src: props.src
}))`
  display: flex;
  align-self: center;
  width: 250px;
  height: 250px;
  object-fit: cover;
  border: ${props => (props.border ? 1 : 0)}px solid black;
`;

export const Title = styled.div`
  font-size: 40px;
  text-align: center;
  color: black;
  border: ${props => (props.border ? 1 : 0)}px solid black;
`;

export const Button = styled.button.attrs(props => ({
  type: "button"
}))`
  width: 100%;
  height: 40px;
  font-size: 15px;
  background-color: #333333;
  color: white;
  :hover {
    background-color: #666666;
    color: white;
  }
  border: ${props => (props.border ? 1 : 0)}px solid black;
`;

export const Link = styled.div`
  font-size: 15px;
  color: #333333;
  padding-top: 20px;
  cursor: pointer;
  border: ${props => (props.border ? 1 : 0)}px solid black;
`;

export const Error = styled.div`
  font-size: 15px;
  color: red;
  padding-top: 10px;
  cursor: pointer;
  border: ${props => (props.border ? 1 : 0)}px solid black;
`;
export const Spinner = styled.div`
  padding-top: 20px;
  align-self: center;
  border: ${props => (props.border ? 1 : 0)}px solid black;
`;
