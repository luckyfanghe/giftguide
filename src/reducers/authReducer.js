// @flow
import types from "../actions/types";

const initialState = {
  sidebarOpened: true,
  sidebarStyle: "login",
  authState: ""
};
function authReducer(state = initialState, action) {
  switch (action.type) {
    case types.AUTH.OPENSIDEBAR:
      return {
        ...state,
        sidebarOpened: true,
        sidebarStyle: action.sidebarStyle
      };
    case types.AUTH.CLOSESIDEBAR:
      return {
        ...state,
        sidebarOpened: false,
        sidebarStyle: action.sidebarStyle
      };
    case types.AUTH.DOINGAUTH:
      return {
        ...state,
        authState: action.state
      };
    default:
      return {
        ...state
      };
  }
}

export default authReducer;
