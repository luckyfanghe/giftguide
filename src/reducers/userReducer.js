// @flow
import types from "../actions/types";

const initialState = {
  user: null,
  userState: "",
  userStateDes: ""
};
function userReducer(state = initialState, action) {
  switch (action.type) {
    case types.USER.UPDATE:
      return {
        ...state,
        user: action.userInfo
      };
    case types.USER.LOGINFAILED:
      return {
        ...state,
        userState: "login failed",
        userStateDes: action.state
      };
    case types.USER.LOGINSUCCESS:
      return {
        ...state,
        userState: "login success",
        userStateDes: ""
      };
    case types.USER.SIGNUPFAILED:
      return {
        ...state,
        userState: "signup failed",
        userStateDes: action.state
      };
    case types.USER.SIGNUPSUCCESS:
      return {
        ...state,
        userState: "signup success",
        userStateDes: ""
      };
    default:
      return {
        ...state
      };
  }
}

export default userReducer;
