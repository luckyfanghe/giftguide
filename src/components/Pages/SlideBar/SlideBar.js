import React from "react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { Link, withRouter } from "react-router-dom";
import { Icon } from "@fickleinfo/react-icons";
import { Form, Button, FormControl } from "react-bootstrap";

import * as S from "./styled";
import logo from "../../../assets/images/Logo-large.webp";
import "../../Styles/style.css";
import {
  openSidebar,
  closeSidebar,
  doingAuth
} from "../../../actions/authActions";
import { loginUser, signupUser } from "../../../actions/userActions";
import Spinner from "../../UI/Spinner";
import * as routes from "../../../constants/routes";

class SlideBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      validated: false,
      errorState: this.props.userStateDes != "" ? true : false
    };
    this.emailTextInput = React.createRef();
    this.passwordTextInput = React.createRef();
  }

  componentDidMount() {}

  _onChangeEmailHandler = e => {
    this.setState({ errorState: false });
  };

  _onChangePasswordHandler = e => {
    this.setState({ errorState: false });
  };

  _handleSubmit = event => {
    event.preventDefault();
    const form = event.currentTarget;
    if (form.checkValidity() === false) {
      this.setState({
        validated: true
      });
      return false;
    } else {
      this.setState({
        validated: false
      });
      this.props.doingAuth("authenticating");
      if (this.props.sidebarStyle == "login")
        this.props.loginUser({
          email: this.emailTextInput.current.value,
          password: this.passwordTextInput.current.value
        });
      else if (this.props.sidebarStyle == "signup")
        this.props.signupUser({
          email: this.emailTextInput.current.value,
          password: this.passwordTextInput.current.value
        });
    }
  };

  _onSignupHandler = e => {
    e.preventDefault();
    this.props.closeSidebar();
    setTimeout(
      function() {
        this.props.openSidebar("signup");
        this.props.history.push(routes.DASHBOARD_PAGE);
      }.bind(this),
      1000
    );
  };

  _onLoginHandler = e => {
    e.preventDefault();
    this.props.closeSidebar();
    setTimeout(
      function() {
        this.props.openSidebar("login");
        this.props.history.push(routes.DASHBOARD_PAGE);
      }.bind(this),
      1000
    );
  };

  loginComponent = () => {
    return (
      <React.Fragment>
        <S.Link> {process.env.NODE_ENV}</S.Link>
        <S.Link onClick={this._onSignupHandler}>create an account</S.Link>
        <S.Link>login with google or facebook</S.Link>
      </React.Fragment>
    );
  };

  signupComponent = () => {
    return (
      <React.Fragment>
        <S.Link onClick={this._onLoginHandler}>
          already have an accont? log in
        </S.Link>
        <S.Link>signup with google or facebook</S.Link>
      </React.Fragment>
    );
  };

  render() {
    let linkComponent;
    if (this.props.sidebarStyle == "login")
      linkComponent = this.loginComponent();
    else if (this.props.sidebarStyle == "signup")
      linkComponent = this.signupComponent();
    return (
      <React.Fragment>
        <S.Body data-testid="SlideBar">
          <S.Logo src={logo} alt="logo" />
          <S.Title>WELCOME TO GIFT GIUDE</S.Title>
          <Form
            noValidate
            validated={this.state.validated}
            onSubmit={this._handleSubmit}
          >
            <Form.Group controlId="formBasicEmail">
              <Form.Label>Email</Form.Label>
              <Form.Control
                ref={this.emailTextInput}
                type="email"
                required
                onClick={this._onChangeEmailHandler}
              />
            </Form.Group>
            <Form.Group controlId="formBasicPassword">
              <Form.Label>Password</Form.Label>
              <Form.Control
                ref={this.passwordTextInput}
                type="password"
                required
                onClick={this._onChangePasswordHandler}
              />
            </Form.Group>
            <Button variant="primary" type="submit" className="auth-button">
              {this.props.sidebarStyle == "login" ? "LOG IN" : "SIGN UP"}
            </Button>
          </Form>
          <S.Error>
            {this.state.errorState ? this.props.userStateDes : ""}
          </S.Error>
          {linkComponent}
          <S.Spinner>
            {this.props.authState == "authenticating" ? (
              <Spinner width={50} height={50} />
            ) : (
              ""
            )}
          </S.Spinner>
        </S.Body>
      </React.Fragment>
    );
  }
}

const actions = dispatch =>
  bindActionCreators(
    {
      openSidebar,
      closeSidebar,
      loginUser,
      signupUser,
      doingAuth
    },
    dispatch
  );

function mapStateToProps(state) {
  return {
    sidebarOpened: state.authReducer.sidebarOpened,
    sidebarStyle: state.authReducer.sidebarStyle,
    authState: state.authReducer.authState,
    userState: state.userReducer.userState,
    userStateDes: state.userReducer.userStateDes
  };
}

export default connect(mapStateToProps, actions)(withRouter(SlideBar));
