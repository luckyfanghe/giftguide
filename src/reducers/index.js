import { combineReducers } from "redux";
import { connectRouter } from "connected-react-router";

import userReducer from "./userReducer";
import authReducer from "./authReducer";

export default combineReducers({
  userReducer,
  authReducer
});
