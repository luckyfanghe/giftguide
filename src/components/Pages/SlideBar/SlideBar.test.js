import React from "react";
import { render, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import SlideBar from "./SlideBar";

import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { composeWithDevTools } from "redux-devtools-extension";
import { createStore, applyMiddleware, compose } from "redux";
import createSagaMiddleware from "redux-saga";
import reducers from "../../../reducers";
import rootSaga from "../../../sagas";

const sagaMiddleware = createSagaMiddleware();
const enhancers = composeWithDevTools(applyMiddleware(sagaMiddleware));

const store = createStore(reducers, enhancers);

sagaMiddleware.run(rootSaga);

it("renders corrently", () => {
  const { queryByTestId } = render(
    <Provider store={store}>
      <BrowserRouter>
        <SlideBar />
      </BrowserRouter>
    </Provider>
  );

  expect(queryByTestId("SlideBar")).toBeTruthy();
});

describe("SlideBar", () => {
  test("SlideBar snapshot renders ", () => {
    const component = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <SlideBar />
        </BrowserRouter>
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
