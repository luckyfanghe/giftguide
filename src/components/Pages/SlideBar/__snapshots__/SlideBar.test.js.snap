// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlideBar SlideBar snapshot renders  1`] = `
<div
  className="sc-AykKC fBCwXg"
  data-testid="SlideBar"
>
  <img
    alt="logo"
    className="sc-AykKE cTtjYa"
    src="Logo-large.webp"
  />
  <div
    className="sc-AykKF gZseDx"
  >
    WELCOME TO GIFT GIUDE
  </div>
  <form
    className=""
    noValidate={true}
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="formBasicEmail"
      >
        Email
      </label>
      <input
        className="form-control"
        id="formBasicEmail"
        onClick={[Function]}
        required={true}
        type="email"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="formBasicPassword"
      >
        Password
      </label>
      <input
        className="form-control"
        id="formBasicPassword"
        onClick={[Function]}
        required={true}
        type="password"
      />
    </div>
    <button
      className="auth-button btn btn-primary"
      disabled={false}
      type="submit"
    >
      LOG IN
    </button>
  </form>
  <div
    className="sc-AykKI jNvuZI"
  >
    
  </div>
  <div
    className="sc-AykKH cGWaIW"
  >
     
    test
  </div>
  <div
    className="sc-AykKH cGWaIW"
    onClick={[Function]}
  >
    create an account
  </div>
  <div
    className="sc-AykKH cGWaIW"
  >
    login with google or facebook
  </div>
  <div
    className="sc-AykKJ cssEoh"
  >
    
  </div>
</div>
`;
